common/src/chain/upgrades/netupgrade.rs

```diff
@@ -54,25 +55,29 @@ pub trait Activate {
     }
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
 pub enum UpgradeVersion {
     ConsensusUpgrade(ConsensusUpgrade),
     SomeUpgrade,
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
 pub enum ConsensusUpgrade {
-    PoW { initial_difficulty: Compact },
-    PoS,
+    PoW {
+        initial_difficulty: Compact,
+    },
+    PoS {
+        initial_difficulty: Compact,
+        config: PoSChainConfig,
+    },
     DSA,
     IgnoreConsensus,
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
 pub enum RequiredConsensus {
-    // Either genesis or previous block was not PoW
     PoW(PoWStatus),
-    PoS,
+    PoS(PoSStatus),
     DSA,
     IgnoreConsensus,
 }
```

Comment:
These two should probably go.

Reply:
`IgnoreConsensus ` seems pretty useful for unit tests. Otherwise, lot's of boilerplate code has to be written

Reply:
I'm wondering whether we should make the updates generic over this enum, such that a dangerous value like `IgnoreConsensus` can't fall into mainnet. Please create an issue for this.

Reply:
what exactly do you mean by "make the updates generic over this enum" ?

Reply:
#792